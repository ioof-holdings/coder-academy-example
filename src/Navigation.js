import React from "react";
import { Link, Match } from "@reach/router";

const NavLink = ({ text, route, active }) => (
  active ? <span>{text}</span> : <Link to={route}>{text}</Link>
)

const Navigation = () => (
  <nav data-testid="navigation" className="Navigation">
    <Match path="/">
    {({ match }) => <NavLink text="Home" route="/" active={match} />}
    </Match>{" "}
    |{" "}
    <Match path="/todos">
    {({ match }) => <NavLink text="Todos Example" route="/todos" active={match} />}
    </Match>
  </nav>
);

export default Navigation;
