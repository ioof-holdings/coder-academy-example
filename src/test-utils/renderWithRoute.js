import React from "react";
import { render } from "@testing-library/react";
import {
  createHistory,
  createMemorySource,
  LocationProvider
} from "@reach/router";

// this is a handy function that I would utilize for any component
// that relies on the router being in context
const renderWithRoute = (ui, route) => {
    const history = createHistory(createMemorySource(route))
    return {
    ...render(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
};

export default renderWithRoute