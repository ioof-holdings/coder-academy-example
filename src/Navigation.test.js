import React from "react";
import { cleanup, fireEvent } from "@testing-library/react";
import "jest-dom/extend-expect";
import renderWithRoute from "./test-utils/renderWithRoute";
import Navigation from "./Navigation";

afterEach(cleanup);

test("should have home not as link when on home route", async () => {
  const { getByText } = renderWithRoute(<Navigation />, "/");

  const navigationNode = getByText("Home");
  expect(navigationNode).toMatchSnapshot();
});

test("should have home as link when not on home route", async () => {
  const { getByText } = renderWithRoute(<Navigation />, "/not-home");

  const navigationNode = getByText("Home");
  expect(navigationNode).toMatchSnapshot();
});

test("should change route to home route when home link clicked", async () => {
  const { getByText, history } = renderWithRoute(<Navigation />, "/not-home");

  const navigationNode = getByText("Home");
  fireEvent.click(navigationNode);
  expect(history.location.pathname).toBe("/");
});

test("should have todos example not as link when on todos route", async () => {
  const { getByText } = renderWithRoute(<Navigation />, "/todos");

  const navigationNode = getByText("Todos Example");
  expect(navigationNode).toMatchSnapshot();
});

test("should have todos example as link when not on todos route", async () => {
  const { getByText } = renderWithRoute(<Navigation />, "/not-todos");

  const navigationNode = getByText("Todos Example");
  expect(navigationNode).toMatchSnapshot();
});

test("should change route to todos route when todos link clicked", async () => {
  const { getByText, history } = renderWithRoute(<Navigation />, "/not-todos");

  const navigationNode = getByText("Todos Example");
  fireEvent.click(navigationNode);
  expect(history.location.pathname).toBe("/todos");
});
