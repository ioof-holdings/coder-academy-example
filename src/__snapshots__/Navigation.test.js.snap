// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have home as link when not on home route 1`] = `
<a
  href="/"
>
  Home
</a>
`;

exports[`should have home not as link when on home route 1`] = `
<span>
  Home
</span>
`;

exports[`should have todos example as link when not on todos route 1`] = `
<a
  href="/todos"
>
  Todos Example
</a>
`;

exports[`should have todos example not as link when on todos route 1`] = `
<span>
  Todos Example
</span>
`;
